blocks/monitoring/evaluators.py

```diff
@@ -186,33 +223,51 @@ def _compile(self):
             be out-sourced to `ComputationGraph` to deal with it.
 
         """
-        if self.buffer_.accumulation_updates:
+        inputs = []
+        outputs = []
+        updates = None
+        if self.theano_buffer_.accumulation_updates:
             updates = OrderedDict()
-            updates.update(self.buffer_.accumulation_updates)
+            updates.update(self.theano_buffer_.accumulation_updates)
             if self.updates:
                 updates.update(self.updates)
-            self._accumulate_fun = theano.function(
-                self.buffer_.inputs, [],
-                updates=updates)
+            # append?
```

Comment:
@mohammadpz I probably caused a bit of confusion here because of something I said yesterday or the day before, so let me just clarify:
```python
l = l + [new_element]  # Slow! Allocates a new list, I believe this was what I told you to avoid
l.extend(new_element)  # Better, because it extends the list in-place
l += [new_element]  # Equivalent to above, but without the overhead of a function call, and looks nicer
l.append(new_element)  # This is the fastest if you're adding just one element
```
Note that we're often talking about nanoseconds for small lists, so it's not that important, but when dealing with large lists it's worth keeping in mind.